### DISCUSSION TITLE
Why is the model structure of `guide` example different from `mnist` example?

### DISCUSSION THREAD

**Discussion by mhfan** [tracel-ai/burn#2121] [Q&A] [answered]
https://github.com/tracel-ai/burn/blob/main/examples/guide/src/model.rs

vs

https://github.com/tracel-ai/burn/blob/main/examples/mnist/src/model.rs

Which one is better?

Besides, I'm curious are they different from the one of https://github.com/tracel-ai/burn/blob/main/examples/onnx-inference/src/model/mnist.onnx ?

**Comment by laggui** [comment, marked as answer]
The mnist example [was used to train a model for the web](https://github.com/tracel-ai/burn/issues/1976#issuecomment-2211595916) example here: https://github.com/tracel-ai/burn/tree/main/examples/mnist-inference-web

The guide is a simple example to get you started with Burn 🙂 

And the last one differs slightly as it is taken from the PyTorch example to export the model to ONNX: https://github.com/tracel-ai/burn/blob/main/examples/onnx-inference/pytorch/mnist.py
